src/MOI_wrapper.jl

```diff
@@ -543,45 +548,66 @@ function vectorize_block(
 end
 
 function MOI.get(
-    optimizer::Optimizer,
+    model::Optimizer,
     attr::MOI.VariablePrimal,
-    vi::MOI.VariableIndex,
+    x::MOI.VariableIndex,
 )
-    MOI.check_result_index_bounds(optimizer, attr)
-    blk, i, j = varmap(optimizer, vi)
-    return block(MOI.get(optimizer, PrimalSolutionMatrix()), blk)[i, j]
+    MOI.check_result_index_bounds(model, attr)
+    blk, i, j = varmap(model, x)
+    return block(model.X, blk)[i, j]
 end
 
 function MOI.get(
-    optimizer::Optimizer,
+    model::Optimizer,
     attr::MOI.ConstraintPrimal,
     ci::MOI.ConstraintIndex{MOI.VectorOfVariables,S},
 ) where {S<:SupportedSets}
-    MOI.check_result_index_bounds(optimizer, attr)
-    return vectorize_block(
-        MOI.get(optimizer, PrimalSolutionMatrix()),
-        block(optimizer, ci),
-        S,
-    )
+    MOI.check_result_index_bounds(model, attr)
+    return vectorize_block(model.X, block(model, ci), S)
 end
-function MOI.get(optimizer::Optimizer, attr::MOI.ConstraintPrimal, ci::AFFEQ)
-    MOI.check_result_index_bounds(optimizer, attr)
-    return optimizer.b[ci.value]
+
+function MOI.get(
+    model::Optimizer,
+    attr::MOI.ConstraintPrimal,
+    ci::MOI.ConstraintIndex{
+        MOI.ScalarAffineFunction{Cdouble},
+        MOI.EqualTo{Cdouble},
+    },
+)
+    MOI.check_result_index_bounds(model, attr)
+    # TODO(odow): this isn't correct. In Ax = b, it should be Ax, not b.
```

Comment:
Does removing this method work now that we have https://github.com/jump-dev/MathOptInterface.jl/pull/1674 ?

Reply:
Yes, we should be able to do something like that. I avoided making large changes in this PR though.